pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -365,6 +365,9 @@ type Cloud struct {
 	// and then get a second request before we attach the volume
 	attachingMutex sync.Mutex
 	attaching      map[types.NodeName]map[mountDevice]awsVolumeID
+
+	// state of our device allocator
+	deviceAllocator DeviceAllocator
```

Comment:
Will this leak as nodes get added and removed but DeviceAllocators never get cleaned up?